### DISCUSSION TITLE
Allow running different digit models on different number groups

### DISCUSSION THREAD

**Discussion by baudneo** [jomjol/AI-on-the-edge-device#2478] [General]
### The Feature

First and foremost, thank you for your work. This is an amazing piece of kit and solves the meter reading problem very neatly!

#### The ask
It would be nice to be able to specify X digit (or analog) number groups and run a different model on each. If I create 2 groups of digit numbers, let group one use the class 11 continuous model and group 2 use the class 100 model.

This would allow for different styles of digits on the same meter face.

### Example image:
![Screenshot_2023-06-12_20-25-15](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/9c08574c-692f-4262-9619-c52b8ac629c2)

5/6 digits are suitable for class 11 continuous model, while the 6/6 digit would benefit from the class 100 model.

Currently, things work very well and only occasionally it has an issue when the 6/6 digit is read as a different number rather than 'NaN', due to the digit rolling up/down. This may be an edge case, but I think overall, this would be a beneficial addition for other users.

TIA!

**Comment by baudneo** [comment]
Also, being able to set confidence for each ROI would be beneficial.

**Comment by ghost** [comment]
If you setup the ROIs like described in https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/

And especially for dig-class100/dig-cont, it will have a better recognition

![](https://user-images.githubusercontent.com/412645/199028748-c48ef5bb-a8d4-4c77-9faf-763e6cf77351.png)

And if nothing helps https://jomjol.github.io/AI-on-the-edge-device-docs/collect-new-images/ to improve the models.

I think it is not necessary to support multiple models.

**Comment by baudneo** [comment]
Hi, thanks for the advice. I had read the docs before starting this project and I have setup the ROI's as close to the examples as I could. 

I will try training my own model and use my own images but, why wouldnt having the option to use different models on different number groups be a good idea? It would save people like me hours upon hours of reading the docs and training my own models. IMHO, its always good to give users more options to fine tune their setup.


Edit: Here are my digit ROI's
![Screenshot_2023-06-13_19-13-35](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/ce3c2053-345b-4434-8ec0-eafc8690bed6)

**Comment by ghost** [comment]
The top border could be a bit lower. Other is fine I think. 

Why not using two models or more? If you have a look in configuration, it is not easy to setup. It would be much more complex for users. But a model for each would need more memory and here we are limited. It would make the device more unstable. So if one model is able to recognize all digits, we should use only one model.
